lib/nrf_modem_lib/nrf_modem_lib_trace.c

```diff
@@ -68,6 +124,13 @@ static int trace_fragment_write(struct nrf_modem_trace_data *frag)
 			LOG_WRN("trace_backend_write wrote 0 bytes.");
 		}
 
+		/* Prevent division by zero (sometimes `end` == `start`) */
+		if (end != start) {
+			bps = ret / (end - start) * 1000 * 8;
+
+			update_bps(bps);
+		}
+
 		remaining -= ret;
```

Comment:
Happy path?
```suggestion
		remaining -= ret;
		if (end == start) {
		        /* Prevent division by zero (sometimes `end` == `start`) */
		        continue;
		}
		bps = ret / (end - start) * 1000 * 8;

		update_bps(bps);
```

Reply:
I don't like the idea of using `continue` like that (more confusing), but I will try to find a different way to write the code if this implementation looks unsatisfying.